### PR TITLE
DPC-692: 1-800-MEDICARE data model

### PR DESCRIPTION
**Pull request by embh**
**Why**
This work begins setting up the dpc-consent module and database, in preparation for building a 1-800-MEDICARE file ETL process and incorporating suppression into the aggregation process.

**What Changed**
* New:
  * dpc-consent module
  * dpc_consent database
  * ConsentEntity
  * ConsentDAO
  * Migration configuration to create consent table

**Choices Made**
ConsentTest exists to confirm that `db migrate` will run without errors and doesn't test anything else at this time.

**Tickets closed**:
* [DPC-692](https://jiraent.cms.gov/browse/DPC-692)

**Future Work**
* [DPC-693](https://jiraent.cms.gov/browse/DPC-693) - Set up 1-800-MEDICARE ETL
* [DPC-694](https://jiraent.cms.gov/browse/DPC-694) - Modify aggregation to account for beneficiary suppression

**Checklist**

- [x] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [x] Swagger documentation has been updated
- [x] FHIR documentation has been updated
* No changes to API, thus no changes to documentation.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -54,6 +54,14 @@ services:
       - export-volume:/app/data
       - ./jacocoReport/dpc-api:/jacoco-report
 
+  consent:
+    image: ${ECR_HOST:-755619740999.dkr.ecr.us-east-1.amazonaws.com/dpc-consent}:latest
+    environment:
+      - ENV=local
+      - JACOCO=${REPORT_COVERAGE}
+    volumes:
+      - ./jacocoReport/dpc-api:/jacoco-report
```
Comment (nickrobison-usds) [inline docker-compose.yml]: This path should be `dpc-consent`

File: dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+public class ConsentEntity implements Serializable {
+
+    private static final long serialVersionUID = 42L;
+
+    @Id
+    @Column(name = "id", updatable = false, nullable = false, columnDefinition = "uuid")
+    private UUID id;
+
+    @NotNull
+    private String beneficiaryID;
```
Comment (nickrobison-usds) [inline dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java]: Can we strengthen this constraint to be `NotEmpty`?

Comment (nickrobison-usds) [reply dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java]: That's the difference between beneficiaryID and mib/hicn?

File: dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    private String mbi;
```
Comment (nickrobison-usds) [inline dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java]: Should these be NotEmpty as well?

File: dpc-consent/pom.xml [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        <dependency>
+            <groupId>${typesafe.config.groupID}</groupId>
+            <artifactId>typesafe-dropwizard-configuration</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.inject</groupId>
+            <artifactId>guice</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jooq</groupId>
```
Comment (nickrobison-usds) [inline dpc-consent/pom.xml]: Are we using (or planning to use Jooq?)

File: dpc-consent/pom.xml [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            <groupId>${hapi.fhir.groupID}</groupId>
+            <artifactId>hapi-fhir-client</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>gov.cms.dpc</groupId>
+            <artifactId>dpc-macaroons</artifactId>
+            <version>0.4.0-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (nickrobison-usds) [inline dpc-consent/pom.xml]: I think we can pull the macaroons library out.

File: dpc-consent/pom.xml [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+                                <file>/entrypoint.sh</file>
+                                <mode>755</mode>
+                            </permission>
+                        </permissions>
+                    </extraDirectories>
+                </configuration>
+            </plugin>
+
+            <plugin>
+                <groupId>org.jooq</groupId>
```
Comment (nickrobison-usds) [inline dpc-consent/pom.xml]: Jooq here as well?

File: dpc-consent/pom.xml (hunk 4 above)
Comment (embh) [reply dpc-consent/pom.xml]: No, thanks for catching that! Left over from trying to figure out if jOOQ was necessary.

File: dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java (hunk 3 above)
Comment (embh) [reply dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java]: The 1-800-MEDICARE files will have only HICNs initially. Would it make sense for only `hicn` to be `NotEmpty` for now, and we add it to `mbi` later? Or maybe we make them both `NotEmpty` now and revisit when we start receiving data?

Comment (nickrobison-usds) [reply dpc-common/src/main/java/gov/cms/dpc/common/entities/ConsentEntity.java]: We can add the restriction MBI later, good point!

File: dpc-consent/pom.xml [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.dropwizard</groupId>
+            <artifactId>dropwizard-migrations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.dropwizard</groupId>
+            <artifactId>dropwizard-hibernate</artifactId>
+        </dependency>
+        <dependency>
```
Comment (nickrobison-usds) [inline dpc-consent/pom.xml]: Are we using Sundial? Probably need to pull it out.

File: dpc-consent/pom.xml [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+        <version>0.4.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>dpc-consent</artifactId>
+    <name>DPC Consent Service</name>
+
+    <properties>
+        <mainClass>gov.cms.dpc.consent.DPCConsentService</mainClass>
+        <jooq.version>3.11.10</jooq.version>
```
Comment (nickrobison-usds) [inline dpc-consent/pom.xml]: Can we remove this unused variable?

File: dpc-consent/pom.xml [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+        <testResources>
+            <testResource>
+                <directory>${project.basedir}/../src/main/resources</directory>
+            </testResource>
+            <testResource>
+                <directory>src/test/resources</directory>
+            </testResource>
+        </testResources>
+        <plugins>
+            <plugin>
```
Comment (nickrobison-usds) [inline dpc-consent/pom.xml]: We'll also need to add the JPA Metamodel generation, which makes building JPA queries a lot easier. You should be able to copy/paste this block.
```
<plugin>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <!--Locking this dependency version to the one provided by Dropwizard, we'll need to manually update it as we go along-->
                    <annotationProcessorPaths>
                        <annotationProcessorPath>
                            <groupId>org.hibernate</groupId>
                            <artifactId>hibernate-jpamodelgen</artifactId>
                            <version>5.4.2.Final</version>
                        </annotationProcessorPath>
                        <path>
                            <groupId>javax.xml.bind</groupId>
                            <artifactId>jaxb-api</artifactId>
                            <version>2.3.0</version>
                        </path>
                        <path>
                            <groupId>javax.annotation</groupId>
                            <artifactId>javax.annotation-api</artifactId>
                            <version>1.3.1</version>
                        </path>
                    </annotationProcessorPaths>
                </configuration>
            </plugin>
```

File: dpc-consent/pom.xml (hunk 7 above)
Comment (embh) [reply dpc-consent/pom.xml]: I removed Sundial, but was thinking we'd likely use it for scheduling the ETL. Would it be better to trigger it some other way?

**Pull request merged**